src/transforms/remap.rs

```diff
@@ -169,24 +207,25 @@ impl RemapConfig {
         };
         let mut config = CompileConfig::default();
 
-        config.set_custom(enrichment_tables);
+        config.set_custom(enrichment_tables.clone());
         config.set_custom(MeaningList::default());
 
-        compile_vrl(&source, &functions, &state, config)
-            .map_err(|diagnostics| {
-                Formatter::new(&source, diagnostics)
-                    .colored()
-                    .to_string()
-                    .into()
-            })
+        let res = compile_vrl(&source, &functions, &state, config)
+            .map_err(|diagnostics| Formatter::new(&source, diagnostics).colored().to_string())
             .map(|result| {
                 (
                     result.program,
                     Formatter::new(&source, result.warnings).to_string(),
-                    functions,
-                    result.config,
+                    result.config.get_custom::<MeaningList>().unwrap().clone(),
                 )
-            })
+            });
+
+        self.cache
+            .lock()
+            .unwrap()
+            .push(((enrichment_tables, merged_schema_definition), res.clone()));
+
+        res.map_err(Into::into)
```

Comment:
Since both this return and the extraction from the cache above maps the error `into`, would it make sense to store the result in the cache with the error already mapped? i.e.
```suggestion
        let res = res.map_err(Into::into);
        self.cache
            .lock()
            .unwrap()
            .push(((enrichment_tables, merged_schema_definition), res.clone()));

        res
```
(and change the type in the `cache` to match the return value from this function)

Reply:
It will require `vector_common::Error` to be `Clone`. I am not sure if we can add this trait to `Error` definition.

Reply:
Oh, I thought I had read the function had a different error return value. Ignore this, then, the overhead isn't going to be worth it for the error path.